### PR TITLE
Align migration table style to Github Markdown.

### PR DESCRIPTION
**Pull request by etefera**
Also moves migration file up a level.
Fixes https://github.com/GoogleContainerTools/kpt/issues/2119
Demo at https://kpt-dev-ee2a9--table-w5atdb7v.web.app/installation/migration
image

### CODE REVIEW

File: site/installation/README.md [hunk 1]
```diff
@@ -96,4 +96,4 @@ kpt help
 [cloud-sdk]: https://github.com/GoogleCloudPlatform/cloud-sdk-docker
 [linux]: https://github.com/GoogleContainerTools/kpt/releases/download/v1.0.0-alpha.4/kpt_linux_amd64
 [darwin]: https://github.com/GoogleContainerTools/kpt/releases/download/v1.0.0-alpha.4/kpt_darwin_amd64
-[migration guide]: installation/migration/
+[migration guide]: installation/migration
```
Comment (phanimarupaka) [inline site/installation/README.md:99]: so both https://kpt.dev/installation/migration/ and https://kpt.dev/installation/migration works after the change ? I think this needs a broader fix(as a separate PR) as the urls don't work without trailing /

Comment (etefera) [reply site/installation/README.md:99]: Unfortunately, just the latter link (https://kpt.dev/installation/migration) will work.
I agree: this is tracked in a separate issue but is a problem with Docsify's history mode. There are some workaround we can explore but the best fix would be at the Docsify level.

Comment (frankfarzan) [reply site/installation/README.md:99]: Speaking of slashes, the rest of the docs have a leading slash:

/installation/migration

Also, don't we have a CI linter that catches broken links? If not, it should be added soon.

Comment (etefera) [reply site/installation/README.md:99]: We do; the link just isn't broken.

**Review by frankfarzan — APPROVED** [review]

**Pull request merged**